### PR TITLE
refactor: remove giant blank in the middle of status and base-sha and…

### PR DESCRIPTION
**Pull request by amirrmonfared**
Ref: https://github.com/kubernetes/test-infra/issues/28923

Signed-off-by: amirrmonfared <amir.monfared@ginbits.com>

### CODE REVIEW

File: prow/config/config.go [hunk 1]
```diff
@@ -3261,18 +3261,9 @@ func (pc *ProwConfig) mergeFrom(additional *ProwConfig) error {
...
-		if len(humanReadable+suffix) < contextDescriptionMaxLen {
-			for i := 0; i < contextDescriptionMaxLen-len(humanReadable+suffix); i++ {
-				// This looks like a standard space but is U+2001, because GitHub seems to deduplicate normal
-				// spaces in their frontend.
-				suffix = " " + suffix
-			}
-		}
+		suffix = "—" + contextDescriptionBaseSHADelimiter + baseSHA
 	}
+	fmt.Printf("humanReadable: %s, suffix: %s ", humanReadable, suffix)
```
Comment (jmguzik) [inline prow/config/config.go:3266]: why this print is added?

File: prow/config/config.go [hunk 2]
```diff
@@ -3261,18 +3261,9 @@ func (pc *ProwConfig) mergeFrom(additional *ProwConfig) error {
...
-		// The GitHub UI will also trim it on the right side and replace some part of it with '...'. The
-		// API always returns the full string.
-		if len(humanReadable+suffix) < contextDescriptionMaxLen {
-			for i := 0; i < contextDescriptionMaxLen-len(humanReadable+suffix); i++ {
-				// This looks like a standard space but is U+2001, because GitHub seems to deduplicate normal
-				// spaces in their frontend.
-				suffix = " " + suffix
-			}
-		}
+		suffix = "—" + contextDescriptionBaseSHADelimiter + baseSHA
```
[the remark this answers is not in the document]
Comment (jmguzik) [reply prow/config/config.go:3264]: @droslean they failed, the code needs to be adjusted in tests as well. There is a test for it.

[the remark this answers is not in the document]
Comment (amirrmonfared) [reply prow/config/config.go:3264]: Thanks for your review, sure I will adjust the tests as well!

**Comment by petr-muller** [comment]
/hold

I'd like a stronger majority to be in favor of this change for it to land. Currently there are two people in favor of this in https://github.com/kubernetes/test-infra/issues/28923 .

**Comment by jmguzik** [comment]
> /hold
> 
> I'd like a stronger majority to be in favor of this change for it to land. Currently there are two people in favor of this in #28923 .

I did not state it but I would like to hear why it was made like this. It's curious as it seems it's implemented on purpose.

**Comment by amirrmonfared** [comment]
/close 
since the referenced issue is closed and this change breaks some test cases as well so let's put energy on the more important stuff. feel free to ping me and re-open this PR in case we felt we need it.

**Pull request closed without merging**